sklearn/utils/tests/test_validation.py

```diff
@@ -539,3 +540,29 @@ def test_suppress_validation():
     assert_all_finite(X)
     sklearn.set_config(assume_finite=False)
     assert_raises(ValueError, assert_all_finite, X)
+
+
+class DummyMemory(object):
+    def __init__(self):
+        pass
+
+    def cache(self, func):
+        return func
+
+    cachedir = None
+
+
+class WrongDummyMemory(object):
+    def __init__(self):
+        pass
+
+
+def test_check_memory():
+    try:
+        check_memory("TestString")
+        check_memory(None)
+        check_memory(DummyMemory())
```

Comment:
you might want to set cachedir to something different than None in DummyMemory. You can make a check then